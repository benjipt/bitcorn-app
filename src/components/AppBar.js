import React from 'react'
import AddressName from './AddressName'
import SignOutBtn from './SignOutBtn'

export default function AppBar({ address, handleLogout }) {
    return (
        <div className="container-fluid pt-2 pb-2 d-flex flex-row justify-content-between app-bar app-bar-border">
            <AddressName address={ address } />
            <SignOutBtn handleLogout={ handleLogout } />
        </div>
    )
}
