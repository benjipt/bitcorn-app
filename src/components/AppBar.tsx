import AddressName from './AddressName';
import SignOutBtn from './SignOutBtn';

interface AppBarProps {
  address: string;
  handleLogout: () => void;
}

export default function AppBar({ address, handleLogout }: AppBarProps) {
  return (
    <div
      className='container-fluid pt-2 pb-2 d-flex flex-row justify-content-between app-bar app-bar-border'
      data-testid='AppBar-1'>
      <AddressName address={address} />
      <SignOutBtn handleLogout={handleLogout} />
    </div>
  );
}
