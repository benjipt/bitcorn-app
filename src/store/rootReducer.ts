// src/store/rootReducer.ts
import { combineReducers } from '@reduxjs/toolkit';
import userReducer from './slices/userSlice';
import uiReducer from './slices/uiSlice';

const rootReducer = combineReducers({
  user: userReducer,
  ui: uiReducer,
});

export default rootReducer;
