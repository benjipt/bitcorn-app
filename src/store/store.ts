// src/store/store.ts
import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';
import rootReducer from './rootReducer';

const store = configureStore({
  reducer: rootReducer,
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;

export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;
export const useAppDispatch = () => useDispatch<AppDispatch>();

export default store;
