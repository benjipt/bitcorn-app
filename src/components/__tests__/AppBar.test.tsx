// AppBar.test.tsx
import { describe, it, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import renderer, { ReactTestRenderer, act } from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '@/store/store';
import AppBar from '@/components/AppBar';

describe('AppBar', () => {
  it('should render AppBar with AddressName and SignOutBtn', () => {
    // Renders Appbar and child components
    render(
      <Provider store={store}>
        <AppBar address='' />
      </Provider>
    );
    // Grabs components from render
    const AppBarComponent = screen.getByTestId('AppBar-1');
    const AddressNameComponent = screen.getByTestId('AddressName-1');
    const SignOutBtnComponent = screen.getByTestId('SignOutBtn-1');
    // Tests for expected components and content
    expect(AppBarComponent).toBeInTheDocument();
    expect(AddressNameComponent).toBeInTheDocument();
    expect(SignOutBtnComponent).toBeInTheDocument();
  });

  it('passed props behave as expected', () => {
    // Mock props
    const address = 'Jilly';
    // Renders Appbar and child components
    render(
      <Provider store={store}>
        <AppBar address={address} />
      </Provider>
    );
    // Grabs components from render
    const AddressNameComponent = screen.getByTestId('AddressName-1');
    // Tests for expected prop behavior
    expect(AddressNameComponent).toHaveTextContent('Jilly');
  });

  it('matches snapshot', () => {
    let tree: ReactTestRenderer | undefined;
    act(() => {
      tree = renderer.create(
        <Provider store={store}>
          <AppBar address='' />
        </Provider>
      );
    });
    expect(tree!.toJSON()).toMatchSnapshot();
  });
});
