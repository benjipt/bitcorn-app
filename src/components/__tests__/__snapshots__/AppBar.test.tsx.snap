// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppBar matches snapshot 1`] = `
<div
  className="container-fluid pt-2 pb-2 d-flex flex-row justify-content-between app-bar app-bar-border"
  data-testid="AppBar-1"
>
  <div
    className="mt-1 ps-4"
    data-testid="AddressName-1"
  >
    <h4>
      
    </h4>
  </div>
  <div>
    <button
      className="btn btn-outline-dark"
      data-testid="SignOutBtn-1"
      onClick={[Function]}
    >
      Sign Out
    </button>
  </div>
</div>
`;
